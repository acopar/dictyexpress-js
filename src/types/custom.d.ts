declare module '*.woff' {
    const src: string;
    export default src;
}

declare module '*.eot' {
    const src: string;
    export default src;
}

declare module '*.ttf' {
    const src: string;
    export default src;
}
