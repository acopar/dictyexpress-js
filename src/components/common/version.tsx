import React, { ReactElement } from 'react';

export const Version = (): ReactElement => (
    <span className="version">
        {import.meta.env.VITE_APP_VERSION}
        {import.meta.env.VITE_APP_DEPLOY_ENV === 'staging'
            ? `+${(import.meta.env.VITE_APP_COMMIT_SHA as string).substring(0, 8)}`
            : ''}
    </span>
);
