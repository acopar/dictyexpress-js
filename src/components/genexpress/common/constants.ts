export enum LocalStorageKey {
    geneSets = 'geneSets',
    layouts = 'layouts',
}

export enum ModulesKeys {
    timeSeriesAndGeneSelector = 'timeSeriesAndGeneSelector',
    expressionTimeCourses = 'expressionTimeCourses',
    differentialExpressions = 'differentialExpressions',
    gOEnrichment = 'gOEnrichment',
    clustering = 'clustering',
}

export enum LayoutBreakpoint {
    large = 'large',
    mid = 'mid',
    small = 'small',
}
