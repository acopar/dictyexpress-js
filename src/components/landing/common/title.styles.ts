import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const Title = styled.h2`
    font-size: 2.5rem;
    color: #49688d;
    text-align: center;
    padding-bottom: 20px;
    margin: 0;
`;
