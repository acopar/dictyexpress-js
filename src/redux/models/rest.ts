export type BasketAddSamplesResponse = {
    id: string;
    modified: string;
    ignored: number[];
    permitted_organisms: string[];
    permitted_sources: string[];
    conflict_organisms: string[];
    conflict_sources: string[];
};
