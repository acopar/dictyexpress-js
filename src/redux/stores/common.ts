import { ActionCreator, ActionReducerMapBuilder } from '@reduxjs/toolkit';

// eslint-disable-next-line import/prefer-default-export
export const clearStateOnActions = <T>(
    builder: ActionReducerMapBuilder<T>,
    actions: ActionCreator<unknown>[],
    initialState: T,
): void => {
    actions.forEach((action) => {
        builder.addCase(action.toString(), () => initialState);
    });
};
