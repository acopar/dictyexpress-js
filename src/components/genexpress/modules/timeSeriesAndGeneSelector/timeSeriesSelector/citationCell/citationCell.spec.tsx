import { customRender } from 'tests/test-utils';
import { fireEvent, screen } from '@testing-library/react';
import { generateSingleTimeSeries } from 'tests/mock';
import CitationCell from './citationCell';
import { Relation } from '@genialis/resolwe/dist/api/types/rest';
import { DescriptorSchemaSlug } from 'components/genexpress/common/constants';
import { Descriptor } from 'types/application';

const relation = generateSingleTimeSeries(1) as Relation & {
    descriptor: Required<Descriptor[DescriptorSchemaSlug.DictyTimeSeries]>;
};

describe('citationCell', () => {
    let asFragment: () => DocumentFragment;

    beforeEach(() => {
        ({ asFragment } = customRender(<CitationCell data={relation} />));
    });

    it('should render as default snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });

    it('should open citation dialog on click', () => {
        fireEvent.click(screen.getByRole('button', { name: relation.descriptor.citation.name }));

        screen.getByText(
            'dictyExpress: a web-based platform for sequence data management and analytics in Dictyostelium and beyond',
            { exact: false },
        );
        screen.getByText('2017', { exact: false });
        screen.getByText('BMC Bioinformatics', { exact: false });
        expect(
            screen.getByRole('link', { name: relation.descriptor.citation.name }),
        ).toHaveAttribute('href', relation.descriptor.citation.url);
        screen.getByRole('heading', {
            name: `${relation.descriptor.project} ${relation.descriptor.details}`,
        });
    });
});
