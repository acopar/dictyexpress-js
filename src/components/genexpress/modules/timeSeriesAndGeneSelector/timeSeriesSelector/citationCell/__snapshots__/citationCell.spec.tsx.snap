// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`citationCell should render as default snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1g352vr-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiBox-root css-1o4wo1x"
    >
      Rosengarten et. al.
    </span>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
